Sources/web3swift/Web3/Web3+Resolver.swift

```diff
@@ -17,10 +17,13 @@ public class PolicyResolver {
     }
 
     public func resolveAll(for tx: inout CodableTransaction, with policies: Policies = .auto) async throws {
+        // FIXME: This should throw an appropriate error, since this condition is critical to whole function tast
         if tx.from != nil || tx.sender != nil {
             // Nonce should be resolved first - as this might be needed for some
             // tx's gas estimation
             tx.nonce = try await resolveNonce(for: tx, with: policies.noncePolicy)
+        } else {
+            throw Web3Error.valueError(desc: "Could not be resolved with both from and sender are nil")
```

Comment:
The tests are failing because read transactions don't really have a `from` or `sender`. I think that's why this was made a condition originally. 

We could think about, if we even need `resolveAll` in `callContractMethod` because for `read` transactions usually none of the _resolved_ values should be relevant.

Reply:
I'm ok with this. What about you @JeneaVranceanu ?

Reply:
Making some more tests, I think we definitely need to remove `resolveAll` from `callContractMethod`. For example when trying to fetch `balanceOf` from an ERC20 token it often fails to resolve.

Reply:
Agree, let's remove `resolveAll`. Ran all the tests without it `callContractMethod` and all worked fine.